autonomy/scripts/mission/mission_controller.py

```diff
@@ -0,0 +1,580 @@
+            self.details_win.addstr(max_lines, self.details_win.getmaxyx()[1] - 3, "↓", 
+                                  curses.color_pair(COLOR_NORMAL))
+        
+        self.details_win.refresh()
+    
+    def draw_screen(self):
+        """Draw all windows based on current view"""
+        # Always draw header and command bar
+        self.draw_header()
+        self.draw_command_bar()
+        
+        # Draw content based on current view
+        if self.current_view == 'main':
+            self.draw_mission_list()
+            self.draw_status()
+        else:  # details view
+            self.draw_details()
+    
+    def show_message(self, message, delay=1.5):
+        """Show a message in the center of the screen"""
+        h, w = self.stdscr.getmaxyx()
+        y = h // 2
+        x = max(0, (w - len(message)) // 2)
+        
+        # Create a popup window
+        msg_h, msg_w = 3, len(message) + 4
+        msg_win = curses.newwin(msg_h, msg_w, max(0, y - 1), max(0, x - 2))
+        msg_win.clear()
+        msg_win.box()
+        msg_win.addstr(1, 2, message)
+        msg_win.refresh()
+        
+        # Wait before removing
+        curses.napms(int(delay * 1000))
+        
+        # Force redraw
+        self.draw_screen()
+    
+    def set_view(self, view):
+        """Change the current view"""
+        self.current_view = view
+        if view == 'details':
+            self.details_scroll = 0  # Reset scroll position
+        self.draw_screen()
+    
+    def select_next_mission(self):
+        """Select the next mission in the list"""
+        if self.available_missions:
+            self.selected_index = (self.selected_index + 1) % len(self.available_missions)
+            self.draw_mission_list()
+    
+    def select_prev_mission(self):
+        """Select the previous mission in the list"""
+        if self.available_missions:
+            self.selected_index = (self.selected_index - 1) % len(self.available_missions)
+            self.draw_mission_list()
+    
+    def scroll_details_up(self):
+        """Scroll details view up"""
+        if self.details_scroll > 0:
+            self.details_scroll -= 1
+            self.draw_details()
+    
+    def scroll_details_down(self):
+        """Scroll details view down"""
+        max_lines = self.details_win.getmaxyx()[0] - 2  # Account for box
+        if self.details_scroll + max_lines < len(self.details_content):
+            self.details_scroll += 1
+            self.draw_details()
+    
+    def select_mission(self):
+        """Select the currently highlighted mission"""
+        if not self.available_missions:
+            self.show_message("No missions available")
+            return
+            
+        selected_mission = self.available_missions[self.selected_index]
+        
+        try:
+            # Call the select_mission service with the mission name as data
+            response = self.select_service(selected_mission)
```

Comment:
Mission manager seems to be expecting a boolean, which the mission controller isn't sending, therefore it gives me the error "Error: field data is not a bool"

Reply:
Also getting this on the mission manager when the controller attempts to send the selected mission
[ERROR] [1747959393.968178]: incoming connection failed: unable to receive data from sender, check sender's logs for details